### ISSUE TITLE
Add extra tests for jsonString, jsonNumber, moveLeftN' and `instance Traversable ListZipper`

### ISSUE THREAD

**Issue by tonymorris** [data61/fp-course#45]
<a href="https://github.com/thomie"><img src="https://avatars.githubusercontent.com/u/157521?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [thomie](https://github.com/thomie)**
_Monday Jan 26, 2015 at 20:50 GMT_
_Originally opened as https://github.com/NICTA/course/pull/112_

----

Please let me know if you want me to split these up into separate pull requests.

The commit with name "Test laziness of moveLeftN'" depends on #111 for the Functor instance of (Either a). 

----

_**[thomie](https://github.com/thomie)** included the following code: https://github.com/NICTA/course/pull/112/commits_
